mods/ra/maps/allies-06b/allies06b.lua

```diff
@@ -0,0 +1,275 @@
+			Utils.Do(TechLabCams, function(a)
+				Actor.Create("TECH.CAM", true, { Owner = player, Location = a.Location })
+			end)
+
+			if Map.LobbyOption("difficulty") == "easy" then
+				Actor.Create("Camera", true, { Owner = player, Location = Weapcam.Location })
+			end
+		end
+	end)
+end
+
+InfiltrateTechCenter = function()
+	Utils.Do(SovietTechLabs, function(a)
+		Trigger.OnInfiltrated(a, function()
+			if infiltrated then
+				return
+			end
+			infiltrated = true
+			DestroySovietsObj = player.AddPrimaryObjective("Destroy all Soviet buildings and units in the area.")
+			player.MarkCompletedObjective(InfiltrateTechCenterObj)
+		end)
+
+		Trigger.OnCapture(a, function()
+			if not infiltrated then
+				Media.DisplayMessage("Do not capture the tech centers! Infiltrate one with a spy.")
+			end
+		end)
+	end)
+
+	Trigger.OnAllKilledOrCaptured(SovietTechLabs, function()
+		if not player.IsObjectiveCompleted(InfiltrateTechCenterObj) then
+			player.MarkFailedObjective(InfiltrateTechCenterObj)
+		end
+	end)
+end
+
+InfiltrateRef = function()
+	Trigger.OnInfiltrated(Refinery, function()
+		player.MarkCompletedObjective(InfiltrateRefObj)
+	end)
+	Trigger.OnCapture(Refinery, function()
+		player.MarkCompletedObjective(InfiltrateRefObj)
+	end)
+	Trigger.OnKilled(Refinery, function()
+		if not player.IsObjectiveCompleted(InfiltrateRefObj) then
+			player.MarkFailedObjective(InfiltrateRefObj)
+		end
+	end)
+end
+
+Tick = function()
+	if player.HasNoRequiredUnits() then
+		player.MarkFailedObjective(InfiltrateTechCenterObj)
+	end
+
+	if DestroySovietsObj and ussr.HasNoRequiredUnits() then
+		player.MarkCompletedObjective(DestroySovietsObj)
+	end
+end
+
+WorldLoaded = function()
+	player = Player.GetPlayer("Greece")
+	ussr = Player.GetPlayer("USSR")
+
+	Trigger.OnObjectiveAdded(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+
+	Trigger.OnObjectiveCompleted(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+
+	Trigger.OnObjectiveFailed(player, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerLost(player, function()
+		Media.PlaySpeechNotification(player, "MissionFailed")
+	end)
+	Trigger.OnPlayerWon(player, function()
+		Media.PlaySpeechNotification(player, "MissionAccomplished")
+	end)
+
+	InfiltrateTechCenterObj = player.AddPrimaryObjective("Infiltrate one of the Soviet tech centers with a spy.")
+	CaptureRadarDomeObj = player.AddSecondaryObjective("Capture the Radar Dome at the shore.")
+	InfiltrateRefObj = player.AddSecondaryObjective("Infiltrate the Refinery for money.")
+
+	Camera.Position = DefaultCameraPosition.CenterPosition
+
+	if Map.LobbyOption("difficulty") ~= "hard" then
+		Utils.Do(TechLabCams, function(a)
+			Actor.Create("TECH.CAM", true, { Owner = player, Location = a.Location })
+		end)
```

Comment:
if Map.LobbyOption("difficulty") ~= "hard" then
		Utils.Do(TechLabCams, function(a)
			Actor.Create("TECH.CAM", true, { Owner = player, Location = a.Location })
		end)

This could be removed, because it now triggers when the radar dome is captured.